### ISSUE TITLE
Getting a message printed when a column contains a JSON field with a null value

### ISSUE THREAD

**Issue by samebchase** [FCO/Red#438]
I have some columns in PG which are of type `json`. Some fields in this JSON object can be `null`.

```raku
model A {
    has Json $.quux_json_field is column;
}
```

When I run the query, for many rows, the JSON parsing works beautifully, but for some I get an error message as follows:

```
Use of uninitialized value element of type Any in string context.
Methods .^name, .raku, .gist, or .say can be used to stringify it to something meaningful.
  in sub from-json at ... JSON::Fast) l
ine 669
```
Upon taking a closer look, the data on DB is something like:

```javascript
{"baz":null,"foo":"bar"}
```
This is what the row printed on console looks like:
```raku
A.new(quux_json_field => ${:baz (Any), :foo("bar")})
```

I created a small test script where I would parse the string using JSON::Fast, and I did not get the message printed.

```raku
say JSON::Fast::from-json('{"baz":null,"foo":"bar"}');
```

Am I doing something wrong here?

I have never used an ORM before, but I am enjoying using Red. 🤓

Thanks in advance 👍

**Comment by FCO** [comment]
Hi!

I'll need to take a look at it after $DAYJOB. Until there, could you run a test for me, please? Could you try adding a `quietly` before the line that's warning that?

I'll take a look, thanks!

**Comment by samebchase** [comment]
Warnings are still printed with code like:

```raku
for A.^all -> $row {
    quietly {
        say $row;
    }
}
```

then I felt that the $row object creation itself is causing the warning to be printed, so I decided to wrap the entire for loop with a `quietly`.

```raku
quietly {
    for ... -> $row {
    }
}
```

and now I am not getting the warning.

I don't mind using `quietly`, but I wonder if this is not a symptom of something wrong I am doing or something erroneous that is happening.

**Comment by FCO** [comment]
No, sorry, I did not suggest you use quietly on your code. I was just trying to use it to helo finding the bug.

I'm going to take a look now

**Comment by FCO** [comment]
@samebchase I couldn't reproduce :(

```
raku -MRed -MRed::Type::Json -e '
model A {
    has Json $.quux_json_field is column;
}

red-defaults "Pg";

.say for A.^all
'
A.new(quux_json_field => ${:baz(Any), :foo("bar")})
```

Would you mind to share a bit more of your code, please?

Thanks

**Comment by samebchase** [comment]
Hi @FCO, I'll try and get you a code snippet that exhibits this behaviour.

**Comment by FCO** [comment]
@samebchase any luck on golf?

**Comment by samebchase** [comment]
@FCO: hey sorry, I did email you a snippet some time ago.

Let me try again with the latest version of RED/Raku and get back to you.

**Comment by FCO** [comment]
Is that still a thing?

**Comment by samebchase** [comment]
Hey @FCO, I'm on paternity leave. It will take some time for me to test it out again. 😅

Will let you know if I have further issues. Thanks for making Red. It's a great library.

**Comment by FCO** [comment]
Congratulations!

Thank you for answering!

**Comment by samebchase** [comment]
@FCO Hey long time,
This seems to be happening with nested JSON fields.

https://gist.github.com/samebchase/48107b480d771d42bb00771bb4520fdf

**Comment by FCO** [comment]
I think it's finally fixed! could you try it, please?

**Comment by samebchase** [comment]
Amazing, it is fixed!

```
test_json_values.new(j_field => ${:foo(1)}, k_val => Any)
{foo => 1}
test_json_values.new(j_field => Any, k_val => ${:foo(1)})
(Any)
test_json_values.new(j_field => Any, k_val => ${:foo(Any)})
(Any)
```
